packages/combobox/src/Combobox.tsx

```diff
@@ -0,0 +1,277 @@
+		},
+		onInputValueChange: async ({ inputValue }) => {
+			await loadOrFilterOptions(inputValue);
+		},
+		onIsOpenChange: async ({ isOpen, inputValue }) => {
+			if (!isOpen || inputValue) return;
+			await loadOrFilterOptions(inputValue);
+		},
+		stateReducer: (state, actionAndChanges) => {
+			const { type: actionAndChangesType, changes } = actionAndChanges;
+			switch (actionAndChangesType) {
+				case useCombobox.stateChangeTypes.InputBlur:
+					return { inputValue: state.selectedItem ? state.inputValue : '' };
+				default:
+					return changes;
+			}
+		},
+	});
+
+	// Popper state
+	const [refEl, setRefEl] = useState<HTMLDivElement | null>(null);
+	const [popperEl, setPopperEl] = useState<HTMLDivElement | null>(null);
+	const { styles, attributes } = usePopper(refEl, popperEl, {
+		placement: 'bottom-start',
+		modifiers: [{ name: 'offset', options: { offset: [0, 8] } }],
+	});
+
+	const { maxWidth, ...inputStyles } = {
+		...textInputStyles({
+			block,
+			maxWidth: 'xl',
+			invalid,
+			valid,
+		}),
+	};
+
+	return (
+		<div css={{ position: 'relative', maxWidth }} ref={setRefEl}>
+			<Field
+				label={label}
+				required={Boolean(required)}
+				hint={hint}
+				message={message}
+				invalid={invalid}
+				valid={valid}
+				id={inputId}
+			>
+				{(allyProps) => (
+					<div css={{ position: 'relative' }} {...getComboboxProps()}>
+						<input
+							css={[inputStyles, { width: '100%' }]}
+							disabled={disabled}
+							{...allyProps}
+							{...getInputProps()}
+						/>
+						{showDropdownTrigger && (
+							<ComboboxDropdownTrigger
+								disabled={disabled}
+								{...getToggleButtonProps()}
+							/>
+						)}
+						{!showDropdownTrigger && clearable && selectedItem && (
+							<ComboboxClearButton disabled={disabled} onClick={reset} />
+						)}
+					</div>
+				)}
+			</Field>
+			<div
+				ref={setPopperEl}
+				style={styles.popper}
+				{...attributes.popper}
+				css={{ zIndex: 1, width: '100%' }}
+			>
+				<ComboboxList {...getMenuProps()} isOpen={isOpen}>
+					{isOpen ? (
+						<Fragment>
+							{loading ? (
+								<ComboboxListLoading />
+							) : networkError ? (
+								<ComboboxListError />
+							) : (
+								<Fragment>
+									{inputItems.length ? (
+										inputItems.map((item, index) => {
+											const isActiveItem = highlightedIndex === index;
+											return (
+												<ComboboxListItem
+													key={`${item.value}${index}`}
+													isActiveItem={isActiveItem}
+													isInteractive={true}
+													{...getItemProps({ item, index })}
+												>
+													{renderItem(item, inputValue)}
+												</ComboboxListItem>
+											);
+										})
+									) : (
+										<ComboboxListEmptyResults message={emptyResultsMessage} />
+									)}
+								</Fragment>
+							)}
+						</Fragment>
+					) : null}
+				</ComboboxList>
+			</div>
+		</div>
+	);
+}
+
+function debounce<A = unknown, R = void>(
```

Comment:
This function isn't used anywhere. Should it be? Or removed?

Reply:
We can remove it for now